Failure to correctly set ids.store-name=titan_ids can corrupt a Titan Cassandra Issue #418

smcquillan opened the pull request:
To fix Issue #418 added a check for the following:

1. `graph.titan-version` is set and is '1.0.0'
2. `ids.store-name` is set in LOCAL and equal to 'titan_ids'
3. `ids.store-name` is not set in GLOBAL ( as read from KCVSConfiguration )

If all the above are true then set `ids.store-name` in the overwrite modifiable configuration to be `titan_ids`. This then prevents the global configuration inheriting the default value of janusgraph_ids but only in the case where ids.store-name had not already been set.

It will now throw an exception and abort if `graph.titan-version` has been set and `ids.store-name` was not also set to the value `titan_ids` which should help preserve data integrity.

Signed-off-by: Scott McQuillan <scott.mcquillan@uk.ibm.com>

pluradj commented:
Thanks @smcquillan . I think this looks good.

It would be worthwhile to start a documentation item under Section V: Advanced Topics on migrating an existing Titan graph to JanusGraph. In the new doc, you can spell out the requirements for these 2 configuration parameters (which would be easily lost in Chapter 12. Configuration Reference).

pluradj commented:
Another thought: is there any particular reason to require `graph.titan-version` and `ids.store-name`? The expected `ids.store-name` value for a Titan migration is a constant, and with your code fixes, it can be set on the initial startup of JanusGraph.

smcquillan commented:
@pluradj Apologies for the delayed response, have been on vacation last week. There's no particular reason to require both `graph.titan-version` and `ids.store-name` be set because as far as I can see the value of `ids.store-name` would always be `titan_ids` if Titan had created the graph.

If as suggested in your earlier comment we add a new advanced topic on migrating an existing Titan graph to JanusGraph then we could drop the need to explicitly set `ids.store-name` and explain that if you did choose to explicitly set its value then it should be to `titan_ids`. Unless there are any objections I'll look into making these changes this week and push them up into this pull request.

smcquillan commented:
@pluradj I've merged up a changes to remove the restriction that `ids.store-name` must be explicitly
set. So now if `graph.titan-version` **is** set and `ids.store-name` is not already in the global config then the default will become `titan_ids`.

Also added a new section on migrating data from Titan under Section V: Advanced Topics and updated the appendix on upgrading to reference it.

pluradj commented on docs/migrating.txt:
```diff
@@ -0,0 +1,31 @@
+
+=== BerkeleyDB
+
+The BerkeleyDB version has been updated, and it contains changes to the file format stored on disk. This file format change is forward compatible with previous versions of BerkeleyDB, so existing graph data stored with Titan can be read in. However, once the data has been read in with the newer version of BerkeleyDB, those files can no longer be read by the older version. Users are encouraged to  ...
```
@jerryjch are there any specific HBase migration concerns from Titan that should be documented here?

pluradj replied:
You can mention the `storage.hbase.table=titan` as the previous default value, similar to what is in the Cassandra section.

smcquillan replied:
@pluradj I've added a section for HBase which basically just covers the `storage.hbase.table=titan` property and I've included a mention about serialisers which I assume will behave the same with HBase as the did with Cassandra. However, I'm not in a position to test with HBase and so will await feedback from @jerryjch on whether this new section is sufficient.
